plugins/ccgx/fu-ccgx-dock-bb.c

```diff
@@ -0,0 +1,734 @@
+
+		if (self->fw_image_type == FW_IMAGE_TYPE_DUAL_SYMMETRIC || self->fw_image_type == FW_IMAGE_TYPE_DUAL_ASYMMETRIC) {
+			if (self->pd_device_data.fw_mode == FW_MODE_FW1)
+				update_fw_mode = FW_MODE_FW2;
+			else if (self->pd_device_data.fw_mode == FW_MODE_FW2)
+				update_fw_mode = FW_MODE_FW1;
+			else if (self->pd_device_data.fw_mode == FW_MODE_BOOT)
+				update_fw_mode = cyacd_file_info.fw_mode;
+			else {
+				g_set_error (error, FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     "not supported fw mode 0x%x", (guint32)self->pd_device_data.fw_mode);
+				g_warning ("not supported fw mode 0x%x", (guint32)self->pd_device_data.fw_mode);
+				return FALSE;;
+			}
+
+			if (cyacd_file_info.fw_mode != update_fw_mode ||
+			    (self->fw_primary_update_only == TRUE && self->pd_device_data.fw_mode  == FW_MODE_FW2)) {
+				if (handle_count > 1)
+					continue; /* get next handle */
+				else
+					break;
+			}
+		}
+
+		g_debug ("===== Update FW file Info =====");
+		g_debug ("  Silicon ID : 0x%X", cyacd_file_info.silicon_id);
+		g_debug ("  FW Mode :  FW%u", (guint32)cyacd_file_info.fw_mode);
+
+		if (self->fw_image_type == FW_IMAGE_TYPE_DUAL_SYMMETRIC)
+			g_debug ("  FW IMG :  DUAL SYMMETRIC");
+		else if (self->fw_image_type == FW_IMAGE_TYPE_DUAL_ASYMMETRIC)
+			g_debug ("  FW IMG :  DUAL ASYMMETRIC");
+
+		g_debug ("  Version : %u.%u.%u /  0x%02X(%c%c)", 
+			(guint32)cyacd_file_info.app_version.ver.major,
+			(guint32) cyacd_file_info.app_version.ver.minor,
+			(guint32)cyacd_file_info.app_version.ver.build,
+			cyacd_file_info.app_version.ver.type,
+			((cyacd_file_info.app_version.ver.type>>8) & 0xff),
+			((cyacd_file_info.app_version.ver.type) & 0xff));
+
+		/* enter FW Mode */
+		if (!fu_ccgx_hpi_cmd_enter_flash_mode (device, hpi_handle, error)) {
+			g_warning("enter flash mode error");
+			return FALSE;
+		}
+
+		g_usleep (HPI_CMD_ENTER_FLASH_MODE_DELAY_US); /* wait 10 msec */
+
+		/* set up meta data row */
+		if (update_fw_mode == FW_MODE_FW1)
+			row_num = self->pd_device_data.fw1_meta_row_num;
+		else
+			row_num = self->pd_device_data.fw2_meta_row_num;
+
+		row_data = &row_buffer[4];
+
+		/* read meta data */
+		row_size = self->pd_device_data.fw_row_size;
+
+		fu_device_set_status (device,FWUPD_STATUS_DEVICE_READ);
+
+		if (!fu_ccgx_hpi_cmd_read_flash (device, hpi_handle, row_num, row_data, row_size, error)) {
+			g_warning("fw meta data read error ");
+			return FALSE;
+		}
+
+		/* get meta data */
+		metadata = (CCGxMetaData*)&row_data[self->pd_device_data.fw_meta_offset];
+
+		/* clear meta data Valid */
+		metadata->metadata_valid = 0x00;
+
+		/* write meta data again */
+		fu_device_set_status (device,FWUPD_STATUS_DEVICE_ERASE);
+
+		if (!fu_ccgx_hpi_cmd_write_flash (device, hpi_handle, row_num, row_data, row_size,error)) {
+			g_warning ("fw meta data write error");
+			return FALSE;
+		}
+
+		/* read flash data */
+		fu_device_set_status (device,FWUPD_STATUS_DEVICE_WRITE);
+
+		g_debug ("Writing Firmware...");
```

Comment:
since the status gets set to write, I suspect you don't actually need this debug line.

Reply:
It is a just debug message to present it is writing firmware.
It is OK to remove it.